### PR TITLE
speaker: allow speaker to have an IPv6 address

### PR DESCRIPTION
**Pull request by mdlayher**
**What this PR does / why we need it**:

First pass at threading a bit of IPv6 goodness through the speaker.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Updates #7.

**Special notes for your reviewer**:

I tried to make things a bit more extensible and general, and got a head start on plumbing a bit of IPv6 addressing through.  It probably needs more work though.

### CODE REVIEW

**Comment by mdlayher** [comment]
Note that IPv6 link local addresses are a little tricky because they can be the same on multiple links, so you  need to specify an interface as well.  I'll get back to this once I have a more clear picture of what this should look like.  Just looking for first thoughts now.

**Comment by mdlayher** [comment]
A thought: it might make sense just to keep the "node IP" part so I can use https://github.com/google/metallb/blob/master/internal/iface/iface.go#L26 to get the interface used for NDP and the IPv6 link local address as well.  But this would all be implicit on the fact that `METALLB_NODE_IP` determines the IPv4, IPv6, and interface information for the speaker.  Thoughts?

**Comment by danderson** [comment]
CircleCI is failing with a lint complaint:

speaker/main.go:92::warning: declaration of "myIP6" shadows declaration at speaker/main.go:90 (vetshadow)

**Comment by mdlayher** [comment]
Ah, easy enough to fix. What do you think otherwise? Particularly the IPv6 configuration bit. Does kubernetes expose that so we can use it in manifests like IPv4?

File: speaker/main.go [hunk 1]
```diff
@@ -103,31 +132,41 @@ type controller struct {
...
-func newController(myIP net.IP, myNode string, noARP bool) (*controller, error) {
-	var arpCtrl *arpController
-	if !noARP {
-		arpAnn, err := arp.New(myIP)
+type controllerConfig struct {
+	IP4    net.IP
+	IP6    net.IP
+	MyNode string
+
+	DisableARP bool
```
Comment (danderson) [inline speaker/main.go]: Please add a comment that this is for testing only. I do still want to remove it at some point.

**Comment by mdlayher** [comment]
I've revised this to accommodate the fact that node IP might be an IPv4 or IPv6 address, and we should use that to infer the interface and then create the speaker configurations.

**Pull request merged**